hr_course/models/hr_course_schedule.py

```diff
@@ -70,6 +75,57 @@ class HrCourseSchedule(models.Model):
         states={"in_validation": [("readonly", False)]},
     )
 
+    alerted = fields.Boolean()
+
+    @api.model
+    def send_course_notification_email(self):
+        company_id = self.env.context.get("company_id") or self.env.company.id
+        channel = (
+            self.env["res.company"].browse(company_id).course_expiration_channel_id
+        )
+
+        if not channel:
+            _logger.info("no channel found for course expiration alerts")
+            return False
+
+        email_template = self.env.ref(
+            "hr_course.mail_template_validity_reminder"
+        ).with_context(lang=self.env.company.partner_id.lang)
+        email_values = email_template.generate_email(self.ids, ["body_html", "subject"])
+
+        values = email_values[self.id]
+        self.with_context(
+            recipient_ids=channel.channel_last_seen_partner_ids.mapped("partner_email"),
```

Comment:
why? Last Seen is likely not what you want to use here.

Reply:
@gurneyalex  which field would be better for that then? We want to inform users assigned to mail channel, right? afair this was the only field like that on channel.

btw. other proposals from your review addressed, only this one left to clarify.